src/functions.php

```diff
@@ -66,6 +66,21 @@ function captureLastError(?EventHint $hint = null): ?EventId
     return SentrySdk::getCurrentHub()->captureLastError($hint);
 }
 
+/**
+ * Captures a CheckIn to the configured Monitor.
+ *
+ * @param string        $slug                Identifier of the Monitor
+ * @param MonitorConfig $upsertMonitorConfig Configuration of the Monitor
+ * @param CheckInStatus $status              The status of the Monitor
+ * @param CheckIn|null  $previous            A CheckIn that may have preceded the current CheckIn
+ *
+ * @return CheckIn The created CheckIn
+ */
+function captureCheckIn(string $slug, MonitorConfig $upsertMonitorConfig, CheckInStatus $status, ?CheckIn $previous = null): ?CheckIn
```

Comment:
Commenting here, but this applies down the call chain.

- A `MonitorConfig` should be optional. People might create their monitor in the UI; hence no need to require the config object. Would also call the parameter `monitorConfig`.
- We still want to support so-called Heartbeats, a single check-in with status `ok` or `error` and an optional duration set.
- We only need the prior check-in ID, so not sure if we should pass in the whole object. Similar to `captureEvent`, we might return the check-in ID only. We could create a concrete `CheckInId` type.

Some examples of how the API could be used:

```php
$checkInId = Sentry\captureCheckIn(
    slug: 'slug',
    status: CheckInStatus::inProgress()
);


// do stuff

Sentry\captureCheckIn(
    checkInId: $checkInId,
    slug: 'slug',
    status: CheckInStatus::ok()
);
```

```php
$checkInId = Sentry\captureCheckIn(
    slug: 'slug',
    status: CheckInStatus::inProgress(),
    monitorConfig: $monitorConfig,
);

// do stuff

Sentry\captureCheckIn(
    checkInId: $checkInId,
    slug: 'slug',
    status: CheckInStatus::ok()
);
```

```php
Sentry\captureCheckIn(
    slug: 'slug',
    status: CheckInStatus::ok(),
    duration: 10,
);
```